core/state/state_object.go

```diff
@@ -303,18 +335,11 @@ func (s *stateObject) updateTrie(db Database) Trie {
 	if metrics.EnabledExpensive {
 		defer func(start time.Time) { s.db.StorageUpdates += time.Since(start) }(time.Now())
 	}
-	// Retrieve the snapshot storage map for the object
+	// The snapshot storage map for the object
 	var storage map[common.Hash][]byte
-	if s.db.snap != nil {
```

Comment:
Any particular reason to move this storage initialization?

Reply:
Yes, we might not need to do it at all, if there are no pending storage changes

Reply:
That is, if the `value == originStorage`